src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            Debug.Assert(flags == CreateObjectFlags.None);
+
+            Guid pictureIID = IPicture.IID;
+            int hr = Marshal.QueryInterface(externalComObject, ref pictureIID, out IntPtr comObject);
+            if (hr == 0)
+            {
+                return new PictureWrapper(comObject);
+            }
+
+            throw new NotImplementedException();
+        }
+
+        protected override void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal struct IUnknownVtbl
+        {
+            public IntPtr QueryInterface;
+            public IntPtr AddRef;
+            public IntPtr Release;
+        }
+
+        internal unsafe struct IStreamVtbl
+        {
+            public IUnknownVtbl IUnknownImpl;
+            public delegate* unmanaged<IntPtr, byte*, uint, uint*, Interop.HRESULT> Read;
+            public delegate* unmanaged<IntPtr, byte*, uint, uint*, Interop.HRESULT> Write;
+            public delegate* unmanaged<IntPtr, long, SeekOrigin, ulong*, Interop.HRESULT> Seek;
+            public delegate* unmanaged<IntPtr, ulong, Interop.HRESULT> SetSize;
+            public delegate* unmanaged<IntPtr, IntPtr, ulong, ulong*, ulong*, Interop.HRESULT> CopyTo;
+            public delegate* unmanaged<IntPtr, uint, Interop.HRESULT> Commit;
+            public delegate* unmanaged<IntPtr, Interop.HRESULT> Revert;
+            public delegate* unmanaged<IntPtr, ulong, ulong, uint, Interop.HRESULT> LockRegion;
+            public delegate* unmanaged<IntPtr, ulong, ulong, uint, Interop.HRESULT> UnlockRegion;
+            public delegate* unmanaged<IntPtr, out Interop.Ole32.STATSTG, Interop.Ole32.STATFLAG, Interop.HRESULT> Stat;
+            public delegate* unmanaged<IntPtr, IntPtr*, Interop.HRESULT> Clone;
+
+            public static void Fill(IStreamVtbl* vtable, IntPtr fpQueryInteface, IntPtr fpAddRef, IntPtr fpRelease)
+            {
+                vtable->IUnknownImpl = new IUnknownVtbl()
+                {
+                    QueryInterface = fpQueryInteface,
+                    AddRef = fpAddRef,
+                    Release = fpRelease
+                };
+                vtable->Read = &ReadImplementation;
+                vtable->Write = &WriteImplementation;
+                vtable->Seek = &SeekImplementation;
+                vtable->SetSize = &SetSizeImplementation;
+                vtable->CopyTo = &CopyToImplementation;
+                vtable->Commit = &CommitImplementation;
+                vtable->Revert = &RevertImplementation;
+                vtable->LockRegion = &LockRegionImplementation;
+                vtable->UnlockRegion = &UnlockRegionImplementation;
+                vtable->Stat = &StatImplementation;
+                vtable->Clone = &CloneImplementation;
+            }
+
+            [UnmanagedCallersOnly]
+            private static Interop.HRESULT ReadImplementation(IntPtr thisPtr, byte* pv, uint cb, uint* pcbRead)
```

Comment:
All these "Implementation" methods are okay but not exception safe. The general pattern here is below. It is general because there are cases where a try/catch may not be needed or some state can be assumed and steps skipped.

```csharp
[UnmanagedCallersOnly]
private static RET FunctionImpl(IntPtr thisPtr, ARGS)
{
    try
    {
        // Marshal ARGS to a form to pass to managed code
        // Resolve thisPtr to instance
        // Call appropriate method on instance
        // Marshal RET and by-ref ARGS to form to pass back to COM
    }
    catch (Exception e)
    {
         return e.HResult;
    }
    finally
    {
        // Clean up
    }
    return 0; // S_OK
}
```